styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java

```diff
@@ -410,7 +410,7 @@ private static boolean isActiveState(RunStateData runStateData) {
 
     final Backfill backfill;
     try {
-      backfill = storage.runInTransaction(tx -> {
+      backfill = storage.runInTransactionWithRetries(tx -> {
```

Comment:
Not sure we want to retry on conflict here. We might end up writing stale data. If there is another transaction won, this one can give up.

Reply:
Retrying makes sense here and will not cause the writing of "stale" data. Imagine that the "losing" request had arrived some milliseconds later and the previous "winning" transaction had already completed. Then there is no transaction conflict, and the latter request completes as well. That is fine and expected. Retrying transactions here is in effect "delaying" the losing transaction so it ends up ordered after the winning transaction. This is in practice not distinguishable from naturally arriving late.

From the user perspective, there are no ordering guarantees on concurrent requests. This is normal and to be expected in the vast majority of APIs and services. The only way for a user to guarantee an ordering between request A and B against a typical CRUD API like styx is to _wait_ for response A before issuing request B. If the user issues request A and B concurrently without waiting for either response, then the result can be that either A or B "wins". This is normal and expected for most APIs.